### PR TITLE
feat: UI of shortcode actions

### PR DESCRIPTION
**Pull request by nivedin**
### Description
This PR introduces a feature in which authenticated user can view their created shortcodes and can delete, copy shortcode link or open the request from their profile page.

### Checks
- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: packages/hoppscotch-app/components/profile/Shortcode.vue [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+const requestMethodLabels = {
+  get: "text-green-500",
+  post: "text-yellow-500",
+  put: "text-blue-500",
+  delete: "text-red-500",
+  default: "text-gray-500",
+}
+
+const timeStampRef = ref()
+const copyIconRefs = ref("copy")
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/components/profile/Shortcode.vue:91]:
```suggestion
const copyIconRefs = ref<"copy" | "check">("copy")
```

File: packages/hoppscotch-app/components/profile/Shortcode.vue [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+const timestampHovered = useElementHover(timeStampRef)
+
+const parseShortcodeRequest = computed(() =>
+  flow(JSON.parse, translateToNewRequest)(props.shortcode.request)
+)
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/components/profile/Shortcode.vue:96]:
```suggestion
const parseShortcodeRequest = computed(() =>
  pipe(props.shortcode.request, JSON.parse, translateToNewRequest)
)
```

File: packages/hoppscotch-app/components/profile/Shortcode.vue [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  emit("delete-shortcode", codeID)
+}
+
+const requestMethodLabels = {
+  get: "text-green-500",
+  post: "text-yellow-500",
+  put: "text-blue-500",
+  delete: "text-red-500",
+  default: "text-gray-500",
+}
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/components/profile/Shortcode.vue:88]:
```suggestion
} as const
```

**Review by AndrewBastin — CHANGES_REQUESTED** [review]
As discussed on call.

**Comment by liyasthomas** [comment]
@nivedin

1. We should provide a `loading state`, a spinner while fetching the remaining entries in the list (intersection observer action).

2. Limit the table's height to a convenient fixed height (~360px). Let the entries overflow within the table's body.

**Review by AndrewBastin — CHANGES_REQUESTED** [review]
As discussed in the call

**Review by liyasthomas — DISMISSED** [review]
LGTM.

**Review by AndrewBastin — DISMISSED** [review]
lgtm :100:

**Review by liyasthomas — APPROVED** [review]

**Pull request merged**